pkg/kv/dynamodb/store.go

```diff
@@ -455,3 +469,48 @@ func (e *EntriesIterator) Err() error {
 func (e *EntriesIterator) Close() {
 	e.err = kv.ErrClosedEntries
 }
+
+// StartPeriodicCheck performs one check and continues every 'interval' in the background
+func (s *Store) StartPeriodicCheck() {
+	interval := s.params.HealthCheckInterval
+	if interval <= 0 || s.cancel != nil {
+		return
+	}
+	s.cancel = make(chan bool)
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		defer func() { s.cancel = nil }()
+		s.logger.WithField("interval", interval).Debug("Starting DynamoDB health check")
+		// check first and loop for checking every interval
+		s.Check()
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
```

Comment:
1. Use single defer func() { ... }()
2. Don't make and use the channel from places, you may write to a nil channel. It is safe to always create a channel and call close(ch) on close. (like the fact that we use the waitgroup even when we do not start a goroutine)